### DISCUSSION TITLE
Update FiveM Chrome version

### DISCUSSION THREAD

**Discussion by Firou91** [citizenfx/fivem#2616] [Engineering Discussion]
Hi there,

Is it possible to improve the Chrome version on FiveM, so that the entire development community can use the native CSS properties added since?

I've only got one example of an unusable property (which is the one I mainly want to use for my server UIs), but if anyone else finds others, please feel free to add it:

```css
backdrop-filter: blur(30px);
```

__Edit:__
impossible to create a drag and drop system, either in css or js

**Comment by Z3rio** [comment]
You can blur the background, you cant blur the actual game though. As that doesnt exist in the context of an NUI.
Blurring of an ui is not really a new property.

It is too, possible to create a drag and drop system. Both via built in html `cursor: drag` (Something like that)
You can also, ofcourse, do it manually via JS

**Reply by Firou91** [reply]
Thank you for your interest in my issue! Drag-and-drop is a thought that a developer friend shared with me. I thought it was strange not to be able to do it, but yes, it can be done.

**Comment by gtasnail** [comment]
You can use some "hacky" methods to use backdrop-filters such as https://github.com/gtasnail/fivem-glsl/ or by using the pepper plugin for `x-cfx-game-view` object types that captures the game render (just pop this behind any element that uses backdrop-filters)

(obviously this has it's drawbacks such as performance issues but the `x-cfx-game-view` object seems to be the best way to do this though at this moment in time)
```js
const container = document.createElement('div');
container.style.cssText = 'position:fixed; top:50%; left:50%; transform:translate(-50%,-50%); display:flex; gap:20px;'; // just div it

const leftScreen = document.createElement('object');
leftScreen.type = 'application/x-cfx-game-view';
leftScreen.style.width = Math.floor(window.innerWidth / 3)
leftScreen.style.height = Math.floor(window.innerHeight / 3)

const rightScreen = document.createElement('object');
rightScreen.type = 'application/x-cfx-game-view';
rightScreen.style.width = Math.floor(window.innerWidth / 3)
rightScreen.style.height = Math.floor(window.innerHeight / 3)

container.appendChild(leftScreen);
container.appendChild(rightScreen);
document.body.appendChild(container);

window.addEventListener('resize', () => { // reeeesizeee
    const newWidth = Math.floor(window.innerWidth / 3);
    const newHeight = Math.floor(window.innerHeight / 3);
    leftScreen.style.width = rightScreen.style.width = `${newWidth}px`;
    leftScreen.style.height = rightScreen.style.height = `${newHeight}px`;
});
```
![render](https://github.com/user-attachments/assets/29165ec7-a517-4545-b163-ceed5011d67b)
![image](https://github.com/user-attachments/assets/3b7d4bd8-4846-4469-a350-1798a6b8478d)

**Reply by Firou91** [reply]
Thank you for your reply, I'll take into account the possibilities of getting around this problem ^^
I still wonder if there will ever be an update that will fix everything. I think it's a shame not to be able to exploit the full potential of vanilla languages.

**Reply by Z3rio** [reply]
> Thank you for your reply, I'll take into account the possibilities of getting around this problem ^^ I still wonder if there will ever be an update that will fix everything. I think it's a shame not to be able to exploit the full potential of vanilla languages.

Please do understand though, blurring would not "be better" if chrome was updated.
I do believe chrome is fairly up-to-date in FiveM though.

The effect from `backdrop-filter: blur(30px);` would still be the same as `filter: blur(30px;`, just that it effects elements in a different way.
You would still not be able to blur the game / create actual good looking & correct blurs

As for dragging and dropping, that too, definitely works.

**Reply by yunglean4171** [reply]
could someone provide me with an example of rendering players view to nui?

**Comment by vipexv** [comment]
Tailwind CSS V4 Gradients and many more features are broken due to the outdated chromium version, please update :o

**Comment by vipexv** [comment]
![image](https://github.com/user-attachments/assets/3e634ee5-b7af-4bb1-877e-dd371e60b517)

**Comment by YpsilonTM** [comment]
Would be nice so we can indeed use tailwindcss V4 😀

**Reply by przxmus** [reply]
there is a workaround for this, setup tailwind via postcss and add this plugin: "postcss-color-converter" and set colorConverter to hex, the postcss config should look like that:
```js
import colorConverter from "postcss-color-converter";
import tailwindcss from "@tailwindcss/postcss";

export default {
  plugins: [
    tailwindcss(),
    colorConverter({
      outputColorFormat: "hex",
    }),
  ],
};
```

then its working

**Comment by Ehbw** [comment]
A new experiment focused on upgrading CEF to version m144 has been live on the forums experiment group for over a week now.

If you aren't apart of the experiments group already, You can request to join the experiments group [here](https://forum.cfx.re/g/experiments)
